api/types/authentication.go

```diff
@@ -261,23 +255,22 @@ func (c *AuthPreferenceV2) SetAllowLocalAuth(b bool) {
 	c.Spec.AllowLocalAuth = NewBoolOption(b)
 }
 
+// setStaticFields sets static resource header and metadata fields.
+func (c *AuthPreferenceV2) setStaticFields() {
+	c.Kind = KindClusterAuthPreference
+	c.Version = V2
+	if c.Metadata.Name == "" {
+		c.Metadata.Name = MetaNameClusterAuthPreference
+	}
```

Comment:
Sorry about a late comment but it occurred to me this should read just as:
```suggestion
	c.Metadata.Name = MetaNameClusterAuthPreference
```